### PR TITLE
compiler: Avoid generation of delta if there are only prelude steps

### PR DESCRIPTION
**Pull request by geaaru**
Create a build.yaml with only prelude steps will never contain delta from the build layer.

### CODE REVIEW

File: pkg/compiler/types/spec/spec.go [hunk 1]
```diff
@@ -242,7 +242,7 @@ func (cs *LuetCompilationSpec) SetSeedImage(s string) {
 }
 
 func (cs *LuetCompilationSpec) EmptyPackage() bool {
-	return len(cs.BuildSteps()) == 0 && len(cs.GetPreBuildSteps()) == 0 && !cs.UnpackedPackage()
+	return len(cs.BuildSteps()) == 0 && !cs.UnpackedPackage()
```
Comment (mudler) [inline pkg/compiler/types/spec/spec.go:245]: In this case you should also check if `package_dir` is not set

**Review by mudler — CHANGES_REQUESTED** [review]
It's breaking the specific case of `package_dir` set, we should guard from that while doing this change

File: pkg/compiler/types/spec/spec.go (hunk 1 above)
Comment (geaaru) [reply pkg/compiler/types/spec/spec.go:245]: mmm is not handled by UnpackedPackage()?

Comment (geaaru) [reply pkg/compiler/types/spec/spec.go:245]: I confirm: https://github.com/mudler/luet/blob/6305d6a556a797eb782773f1ef79cc9be55068a4/pkg/compiler/types/spec/spec.go#L248

the UnpackedPackage() return true if package_dir is defined.

**Comment by geaaru** [comment]
Can you go ahead with this?

**Review by mudler — APPROVED** [review]

**Pull request merged**